Indent is inconsistent when call expression is used as variable initial value

**Tell us about your environment**

* **ESLint Version:**
master
* **Node Version:**
4
* **npm Version:**
4

**What parser (default, Babel-ESLint, etc.) are you using?**
typescript

**Please show your full configuration:**

```
module.exports = {
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
    "parser": "typescript-eslint-parser",
    "parserOptions": {
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "indent": [
            "error",
            4,
            {
                MemberExpression: 1,
                SwitchCase: 1,
                FunctionDeclaration: {
                    body: 1,
                    parameters: 1
                },
                FunctionExpression: {
                    body: 1,
                    parameters: 1
                },
                CallExpression: {
                    arguments: 1
                },
                VariableDeclarator: 1
            }
        ]
    }
};

```

**What did you do? Please include the actual source code causing the issue.**

```js
// Invalid
const foo = a.b(),
    longName
        = memberFunction(
            'bar'
        );
```

**What did you expect to happen?**
Should be a valid indent. If I move the equal operator to the same line as the variable identifier it is consider valid indentation.
```js
// Valid
const foo = a.b(),
    longName =
        memberFunction(
            'bar'
        );
```

**What actually happened? Please include the actual, raw output from ESLint.**
Invalid indentation.
```
/home/reyad/Workspace/eslint-test/test-indent.ts
  4:1  error  Expected indentation of 8 spaces but found 12  indent
  5:1  error  Expected indentation of 4 spaces but found 8   indent
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1189,9 +1189,11 @@ module.exports = {
             VariableDeclarator(node) {
                 if (node.init) {
                     const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
+                    const tokenAfterOperator = sourceCode.getTokenAfter(equalOperator);
 
                     offsets.ignoreToken(equalOperator);
-                    offsets.ignoreToken(sourceCode.getFirstToken(node.init));
+                    offsets.ignoreToken(tokenAfterOperator);
+                    offsets.matchIndentOf(equalOperator, tokenAfterOperator);
                 }
             },
 

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -3424,6 +3424,133 @@ ruleTester.run("indent", rule, {
                 )
             `
         },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                        (memberFunction(
+                            'bar',
+                            'bar'
+                        ));
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                    (memberFunction(
+                        'bar',
+                        'bar'
+                    ));
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                        memberFunction(
+                            'bar',
+                            'bar'
+                        );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                    memberFunction(
+                        'bar',
+                        'bar'
+                    );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                        = memberFunction(
+                            'bar',
+                            'bar'
+                        );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                    = memberFunction(
+                        'bar',
+                        'bar'
+                    );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                        ('fff');
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                    ('fff');
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                        = ('fff');
+
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                    = ('fff');
+
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                        (
+                            'fff'
+                        );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                    (
+                        'fff'
+                    );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                        =(
+                            'fff'
+                        );
+            `
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                    =(
+                        'fff'
+                    );
+            `
+        },
+
 
         //----------------------------------------------------------------------
         // Ignore Unknown Nodes
@@ -7632,6 +7759,80 @@ ruleTester.run("indent", rule, {
                     baz
             `,
             errors: expectedErrors([[2, 4, 2, "Identifier"], [3, 4, 6, "Identifier"]])
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                    = (memberFunction(
+                            'bar',
+                            'bar'
+                        ));
+            `,
+            output: unIndent`
+                const foo = a.b(),
+                    longName
+                    = (memberFunction(
+                        'bar',
+                        'bar'
+                    ));
+            `,
+            errors: expectedErrors([[4, 8, 12, "String"], [5, 8, 12, "String"], [6, 4, 8, "Punctuator"]])
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName =
+                    (memberFunction(
+                            'bar',
+                            'bar'
+                        ));
+            `,
+            output: unIndent`
+                const foo = a.b(),
+                    longName =
+                    (memberFunction(
+                        'bar',
+                        'bar'
+                    ));
+            `,
+            errors: expectedErrors([[4, 8, 12, "String"], [5, 8, 12, "String"], [6, 4, 8, "Punctuator"]])
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                        =memberFunction(
+                            'bar',
+                            'bar'
+                    );
+            `,
+            output: unIndent`
+                const foo = a.b(),
+                    longName
+                        =memberFunction(
+                            'bar',
+                            'bar'
+                        );
+            `,
+            errors: expectedErrors([[6, 8, 4, "Punctuator"]])
+        },
+        {
+            code: unIndent`
+                const foo = a.b(),
+                    longName
+                        =(
+                        'fff'
+                        );
+            `,
+            output: unIndent`
+                const foo = a.b(),
+                    longName
+                        =(
+                            'fff'
+                        );
+            `,
+            errors: expectedErrors([[4, 12, 8, "String"]])
         },
 
         //----------------------------------------------------------------------

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -3428,7 +3428,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName =
-                        (memberFunction(
+                        (baz(
                             'bar',
                             'bar'
                         ));
@@ -3438,7 +3438,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName =
-                    (memberFunction(
+                    (baz(
                         'bar',
                         'bar'
                     ));
@@ -3448,7 +3448,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName =
-                        memberFunction(
+                        baz(
                             'bar',
                             'bar'
                         );
@@ -3458,7 +3458,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName =
-                    memberFunction(
+                    baz(
                         'bar',
                         'bar'
                     );
@@ -3468,7 +3468,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName
-                        = memberFunction(
+                        = baz(
                             'bar',
                             'bar'
                         );
@@ -3478,7 +3478,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName
-                    = memberFunction(
+                    = baz(
                         'bar',
                         'bar'
                     );
@@ -7764,7 +7764,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName
-                    = (memberFunction(
+                    = (baz(
                             'bar',
                             'bar'
                         ));
@@ -7772,7 +7772,7 @@ ruleTester.run("indent", rule, {
             output: unIndent`
                 const foo = a.b(),
                     longName
-                    = (memberFunction(
+                    = (baz(
                         'bar',
                         'bar'
                     ));
@@ -7783,7 +7783,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName =
-                    (memberFunction(
+                    (baz(
                             'bar',
                             'bar'
                         ));
@@ -7791,7 +7791,7 @@ ruleTester.run("indent", rule, {
             output: unIndent`
                 const foo = a.b(),
                     longName =
-                    (memberFunction(
+                    (baz(
                         'bar',
                         'bar'
                     ));
@@ -7802,7 +7802,7 @@ ruleTester.run("indent", rule, {
             code: unIndent`
                 const foo = a.b(),
                     longName
-                        =memberFunction(
+                        =baz(
                             'bar',
                             'bar'
                     );
@@ -7810,7 +7810,7 @@ ruleTester.run("indent", rule, {
             output: unIndent`
                 const foo = a.b(),
                     longName
-                        =memberFunction(
+                        =baz(
                             'bar',
                             'bar'
                         );

